code/modules/awaymissions/capture_the_flag.dm

```diff
@@ -342,12 +331,20 @@
 
 /obj/machinery/capture_the_flag/proc/reset_the_arena()
 	var/area/A = get_area(src)
+	var/list/ctf_object_typecache = typecacheof(list(
```

Comment:
Make it global

Reply:
it doesn't need to exist after the proc is over

Reply:
Make it global and only initialize it when proc is called, but keep it stored so it won't be initialized again if the proc is called again.

Reply:
CTF almost never happens so I'm not worried about this really, but feel free to PR the tweak yourself

Reply:
it doesn't need to exist, we don't have to pay a permanent cost of this for something that barely ever happens and doesn't cost all that much to generate when needed